operator/scripts/create-olm-test-resources.sh

```diff
@@ -27,23 +29,28 @@ spec:
       interval: 10m
 EOF
 
-cat << EOF >> $SCRIPT_DIR/../olm/testing-resources/operatorgroup.yaml
+
+OPERATOR_GROUP_FILE=$SCRIPT_DIR/../olm/testing-resources/operatorgroup.yaml
+
+cat << EOF >> $OPERATOR_GROUP_FILE
 kind: OperatorGroup
 apiVersion: operators.coreos.com/v1
 metadata:
-  name: og-single
-  namespace: default
+  name: og
 spec:
-  targetNamespaces:
-  - default
 EOF
 
+IFS=', ' read -r -a array <<< "$TARGET_NAMESPACES"
+for element in "${array[@]}"
+do
+    yq ea -i ".spec.targetNamespaces += [\"$element\"]" $OPERATOR_GROUP_FILE
```

Comment:
`spec.TargetNamespaces` seems to be kinda [deprecated](https://docs.openshift.com/container-platform/4.13/operators/understanding/olm/olm-understanding-operatorgroups.html#olm-operatorgroups-target-namespace_olm-understanding-operatorgroups)? But maybe we don't even need it (specifying multiple namespaces here), see my comment above.

Reply:
I would be in favor of separating that off as a another issue around supporting the selector - in general that will be a problem as the sdk nor our logic is honoring that.

Reply:
olm does manage the watching of namespaces for the selector, and updates the operator deployment from there.

I see two issues with that (not related to keycloak):
- the docuemtation is wrong, you cannot directly specify the label under the selector, it needs to be under a matchLabels stanza - if you try it as documented it does end up watching all namespaces.  I'll open something for that.  
- if nothing matches the label olm won't update the deployment because it can't distinguish between watching nothing and watching everything.  This seems like odd behavior as you'll end up still watching one or more namespaces that you don't intend to.

So on the script changes if you want I can either backout everything and we won't bother with those changes for now, or it could be updated to be based upon a label, or a label and target namespaces (such that the script adds the label to the targets).

Reply:
Thanks for double checking!

> So on the script changes if you want I can either backout everything and we won't bother with those changes for now, or it could be updated to be based upon a label, or a label and target namespaces (such that the script adds the label to the targets).

I'd keep only what's strictly necessary for testing these two scenarios in the CI:
* Operator watching and being installed in a single namespace.
* Operator watching all namespaces, i.e. empty watched namespaces.
* And maybe one more scenario that we currently miss. That the Operator can watch a list of namespaces (MultiNamespace mode). I think we technically "support" this now. But we could add this test scenario later as a follow-up.

Reply:
> Which means we can keep everything in the scripts as we'd use targetNamespaces for the last scenario.

If I understand you correctly you mean just leave it as is in the pr and don't try to switch to a label or namespace/label based mechanism.